src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareSaveScmpOperation.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            Validate.IsNotNull("parameters", parameters);
+            Validate.IsNotNull("parameters.ScmpFilePath", parameters.ScmpFilePath);
+            this.Parameters = parameters;
+            this.SourceConnectionString = SchemaCompareUtils.GetConnectionString(sourceConnInfo, parameters.SourceEndpointInfo.DatabaseName);
+            this.TargetConnectionString = SchemaCompareUtils.GetConnectionString(targetConnInfo, parameters.TargetEndpointInfo.DatabaseName);
+            this.OperationId = Guid.NewGuid().ToString();
+        }
+
+
+        public void Execute(TaskExecutionMode mode = TaskExecutionMode.Execute)
+        {
+            if (this.CancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException(this.CancellationToken);
+            }
+
+            try
+            {
+                SchemaCompareEndpoint sourceEndpoint = SchemaCompareUtils.CreateSchemaCompareEndpoint(this.Parameters.SourceEndpointInfo, this.SourceConnectionString);
+                SchemaCompareEndpoint targetEndpoint = SchemaCompareUtils.CreateSchemaCompareEndpoint(this.Parameters.TargetEndpointInfo, this.TargetConnectionString);
+
+                SchemaComparison comparison = new SchemaComparison(sourceEndpoint, targetEndpoint);
+
+                if (Parameters.ExcludedSourceObjects != null)
+                {
+                    foreach (var sourceObj in this.Parameters.ExcludedSourceObjects)
+                    {
+                        SchemaComparisonExcludedObjectId excludedObjId = CreateExcludedObject(sourceObj);
+                        if (excludedObjId != null)
+                        {
+                            comparison.ExcludedSourceObjects.Add(excludedObjId);
+                        }
+                    }
+                }
+
+                if (Parameters.ExcludedTargetObjects != null)
+                {
+                    foreach (var targetObj in this.Parameters.ExcludedTargetObjects)
+                    {
+                        SchemaComparisonExcludedObjectId excludedObjId = CreateExcludedObject(targetObj);
+                        if (excludedObjId != null)
+                        {
+                            comparison.ExcludedTargetObjects.Add(excludedObjId);
+                        }
+                    }
+                }
+
+                if (this.Parameters.DeploymentOptions != null)
+                {
+                    comparison.Options = SchemaCompareUtils.CreateSchemaCompareOptions(this.Parameters.DeploymentOptions);
+                }
+
+                comparison.SaveToFile(this.Parameters.ScmpFilePath, true);
+
+            }
+            catch (Exception e)
+            {
+                ErrorMessage = e.Message;
+                Logger.Write(TraceEventType.Error, string.Format("Schema compare save settings operation {0} failed with exception {1}", this.OperationId, e.Message));
```

Comment:
Can we log the exception object? We lose all of the key information like the stack trace and inner exceptions when we log just the message.

Reply:
done